lib/samples/set_reference_scale/set_reference_scale_sample.dart

```diff
@@ -0,0 +1,225 @@
+                      _scale = newScale!;
+                      _map.referenceScale = _scale;
+                    });
+                  },
+                  items: _referenceScaleList,
+                ),
+                Text('Apply Reference Scale to Layers',
+                    style: Theme.of(context).textTheme.titleMedium),
+
+                // Add a list of checkboxes for selecting feature layers that will honor the reference scale.
+                Column(
+                  children: [
+                    // Create a checkbox for each feature layer.
+                    for (final layer in _allFeatureLayers)
+                      CheckboxListTile(
+                        value: _selectedFeatureLayers.contains(layer),
+                        onChanged: (value) {
+                          setNewState(() {
+                            // Update the selected feature layers list.
+                            if (value ?? false) {
+                              _selectedFeatureLayers.add(layer);
+                            } else {
+                              _selectedFeatureLayers.remove(layer);
+                            }
+                          });
+
+                          // Get the matching layer from the map.
+                          var matchingLayer = _map.operationalLayers
+                              .where((element) => element.name == layer)
+                              .first as FeatureLayer;
+
+                          // Set the layer property based on the checkbox value.
+                          _selectedFeatureLayers.contains(matchingLayer.name)
+                              ? matchingLayer.scaleSymbols = true
+                              : matchingLayer.scaleSymbols = false;
+                        },
+                        // Set the title of the checkbox to the layer name.
+                        title: Text(layer),
+                      ),
+                  ],
+                ),
+                // Add text to display the current map scale.
+                Center(
+                  child: Text('Map Scale',
+                      style: Theme.of(context).textTheme.titleMedium),
+                ),
+                Text(
+                  '1:${_mapViewController.scale.toInt().toString().replaceAllMapped(reg, mathFunc)}',
+                ),
+                ElevatedButton(
+                  onPressed: () {
+                    // Set the map scale to the reference scale and close the settings dialog.
+                    _mapViewController.setViewpointScale(
+                        scale: _map.referenceScale);
+                    Navigator.pop(context);
+                  },
+                  child: const Text('Set to Reference Scale'),
+                ),
+              ],
+            ),
+          ),
+        ),
+      );
+    });
+  }
+
+  // Create a function to format the scale.
+  String Function(Match) mathFunc = (Match match) => '${match[1]},';
```

Comment:
`mathFunc` should be `matchFunc`, although to make it more descriptive it could maybe be `matchFormatter`. And `reg` could also use a more descriptive name, such as `digitGroupRegex` or something like that.